### ISSUE TITLE
SFTTrainer training stops early?

### ISSUE THREAD

**Issue by ka4on** [huggingface/trl#1008]
It seems it just happen today. Has anyone met the error where the training stops early?  For example, it should have run 1000 steps but stops after 300 steps, and giving a output like:

`TrainOutput(global_step=100, training_loss=1.2991660499572755, metrics={'train_runtime': 855.9696, 'train_samples_per_second': 1.869, 'train_steps_per_second': 0.117, 'total_flos': 6.50352940548096e+16, 'train_loss': 1.2991660499572755, 'epoch': 0.16})`

I did my training on Colab, and i don't know if it is an internet issue. But my previous training never meets this kind of issue.

**Comment by ka4on** [comment]
seems caused by `packing = True`

**Comment by lvwerra** [comment]
Yes, when you pack it can happen that you end up with fewer samples if the average length of samples is shorter than the sequence length.
